src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -1906,32 +1906,44 @@ def list_av_sets(cmd, resource_group_name=None):
 
 # region VirtualMachines BootDiagnostics
 def disable_boot_diagnostics(cmd, resource_group_name, vm_name):
-    vm = get_vm_to_update(cmd, resource_group_name, vm_name)
-    diag_profile = vm.diagnostics_profile
-    if not (diag_profile and diag_profile.boot_diagnostics and diag_profile.boot_diagnostics.enabled):
+    from .aaz.latest.vm import Update as UpdateVM, Show as GetVM
+    vm = GetVM(cli_ctx=cmd.cli_ctx)(command_args={
+        'resource_group': resource_group_name,
+        'vm_name': vm_name
+    })
+    diag_profile = False if 'diagnosticsProfile' not in vm else vm['diagnosticsProfile']
+    if not (diag_profile and diag_profile['bootDiagnostics'] and diag_profile['bootDiagnostics']['enabled']):
         return
 
-    diag_profile.boot_diagnostics.enabled = False
-    diag_profile.boot_diagnostics.storage_uri = None
-    set_vm(cmd, vm, ExtensionUpdateLongRunningOperation(cmd.cli_ctx, 'disabling boot diagnostics', 'done'))
+    boot_diag = {'enabled': False, 'storageUri': None}
+    ExtensionUpdateLongRunningOperation(cmd.cli_ctx, 'enabling boot diagnostics', 'done')(
+        UpdateVM(cli_ctx=cmd.cli_ctx)(command_args={
+            'resource_group': resource_group_name,
+            'vm_name': vm_name,
+            'diagnostics_profile': {
+                'bootDiagnostics': boot_diag
```

Comment:
It should be snake case instead of camel case, since the input of update command is snake case:
https://github.com/Azure/azure-cli/blob/2690e056ec67d59d7a8af88e6ba1706769ffe7c4/src/azure-cli/azure/cli/command_modules/vm/aaz/latest/vm/_update.py#L358-L362
```suggestion
                'boot_diagnostics': boot_diag
```